### DISCUSSION TITLE
How to overcome virtual keyboard with uc properties?

### DISCUSSION THREAD

**Discussion by breadacker** [seleniumbase/SeleniumBase#3197] [Q&A] [answered]
def click_shift(sb):
    """
    Clicks the Shift key on the virtual keyboard.
    """
    try:
        shift_key_xpath = "//div[contains(@class, 'cdk-overlay-container')]//button[contains(@class, 'mat-keyboard-key-shift')]"
        sb.wait_for_element_clickable(shift_key_xpath, by=By.XPATH, timeout=5)
        sb.driver.click(shift_key_xpath, by=By.XPATH)
        sb.sleep(0.2)
        print("Clicked on Shift key.")
    except TimeoutException:
        print("Timeout: Shift key not found or not clickable.")
    except NoSuchElementException:
        print("No such element: Shift key not found.")
    except ElementClickInterceptedException:
        print("Click intercepted: Could not click Shift key.")
    except Exception as e:
        print(f"Unexpected error clicking Shift key: {e}")

**Comment by mdmintz** [comment, marked as answer]
If you're seeing `ElementClickInterceptedException`, then use `js_click(css_selector)`.
You might also be able to use `PyAutoGUI` for interaction.

**Reply by breadacker** [reply]
thanks a lot it even works faster than sb.click
I have 1 more question....
//sb.driver.type("input[formcontrolname='username']", "text@gmail.com")  # Update with your email
    sb.driver.uc_click("input[formcontrolname='username']")
    sb.driver.uc_gui_write("text@gmail.com")
    
Instead of using sb.type to mitigate the risk of bot detection, I tried using uc_click and uc_gui_write respectively; however, it didnt work

**Reply by mdmintz** [reply]
If there's real-time bot-scanning involved, see [examples/raw_antibot_login.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_antibot_login.py).
(It uses the `uc_gui` `PyAutoGUI` methods for actions.)

**Reply by breadacker** [reply]
Thanks a lot, you are a life saver 
What are the differences between driver and base_case methods? (eg: driver.click vs self.click)

**Reply by mdmintz** [reply]
They're similar, but the BaseCase methods are more extensive, with more integrations (Eg: Highlighting actions in Demo Mode)
